src/main/java/frc/robot/commandgroups/Outtake.java

```diff
@@ -0,0 +1,34 @@
+package frc.robot.commandgroups;
+
+import edu.wpi.first.wpilibj2.command.ConditionalCommand;
+import edu.wpi.first.wpilibj2.command.ParallelCommandGroup;
+import frc.robot.Constants;
+import frc.robot.subsystems.conveyor.Conveyor;
+import frc.robot.subsystems.conveyor.commands.Convey;
+import frc.robot.subsystems.conveyor.commands.FlapDefaultCommand;
+import frc.robot.subsystems.intake.Intake;
+import frc.robot.subsystems.intake.commands.IntakeCargo;
+import frc.robot.subsystems.shooter.Shooter;
+import frc.robot.subsystems.shooter.commands.Shoot;
+
+import java.util.OptionalDouble;
+import java.util.function.BooleanSupplier;
+import java.util.function.DoubleSupplier;
+
+public class Outtake extends ParallelCommandGroup {
+
+    public Outtake(Intake intake,
+                   Conveyor conveyor,
+                   Shooter shooter,
+                   BooleanSupplier condition) {
+        addCommands(
+                new FlapDefaultCommand(conveyor, () -> Conveyor.FlapMode.Open),
+                new Convey(conveyor, Constants.Conveyor.DEFAULT_POWER * (condition.getAsBoolean() ? 1 : -1)),
+        new DynamicConditionalCommand(
+                    condition,
+                    new Shoot(shooter, () -> 8, OptionalDouble.of(Constants.Shooter.OUTTAKE_POWER)),
+                    new IntakeCargo(intake, -Constants.Intake.DEFAULT_POWER)
+                )
```

Comment:
`Constants.Conveyor.DEFAULT_POWER * (condition.getAsBoolean() ? 1 : -1)` won't work (because you pass a number and not a supplier), there are 2 options: the first one is, of course, using supplier, and the other one is adding `.alongWith()` to `Shoot` & `IntakeCargo`